# HG changeset patch
# User Lisandro Dalcin <dalcinl@gmail.com>
# Date 1239046307 10800
# Node ID 1072bb7a4b85851deb90eeb6e6ded15abc2b86e9
# Parent  ea5b7e6e9db7dcd2f07affbd3f985e8b53b5b335
use PY_FORMAT_SIZE_T to format Py_ssize_t values

diff -r ea5b7e6e9db7 -r 1072bb7a4b85 Cython/Compiler/ExprNodes.py
--- a/Cython/Compiler/ExprNodes.py	Mon Apr 06 14:08:13 2009 -0300
+++ b/Cython/Compiler/ExprNodes.py	Mon Apr 06 16:31:47 2009 -0300
@@ -2133,7 +2133,7 @@
             check = stop
         if check:
             code.putln("if (unlikely((%s) != %d)) {" % (check, target_size))
-            code.putln('PyErr_Format(PyExc_ValueError, "Assignment to slice of wrong length, expected %%d, got %%d", %d, (%s));' % (
+            code.putln('PyErr_Format(PyExc_ValueError, "Assignment to slice of wrong length, expected %%"PY_FORMAT_SIZE_T"d, got %%"PY_FORMAT_SIZE_T"d", (Py_ssize_t)%d, (Py_ssize_t)(%s));' % (
                         target_size, check))
             code.putln(code.error_goto(self.pos))
             code.putln("}")
diff -r ea5b7e6e9db7 -r 1072bb7a4b85 Cython/Compiler/ModuleNode.py
--- a/Cython/Compiler/ModuleNode.py	Mon Apr 06 14:08:13 2009 -0300
+++ b/Cython/Compiler/ModuleNode.py	Mon Apr 06 16:31:47 2009 -0300
@@ -425,6 +425,7 @@
         code.putln("  typedef int Py_ssize_t;")
         code.putln("  #define PY_SSIZE_T_MAX INT_MAX")
         code.putln("  #define PY_SSIZE_T_MIN INT_MIN")
+        code.putln("  #define PY_FORMAT_SIZE_T \"\"")
         code.putln("  #define PyInt_FromSsize_t(z) PyInt_FromLong(z)")
         code.putln("  #define PyInt_AsSsize_t(o)   PyInt_AsLong(o)")
         code.putln("  #define PyNumber_Index(o)    PyNumber_Int(o)")
